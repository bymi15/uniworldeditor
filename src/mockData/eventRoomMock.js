const mockEventRoom = {
  title: 'mockTitle',
  eventDate: 'mockDate',
  meetingTables: [],
  background: 'mockBackground',
  scene: 'mockScene',
};

const mockEventRoomUpdate = {
  title: 'newMockTitle',
  eventDate: 'newMockDate',
  meetingTables: [],
  background: 'newMockBackground',
  scene: 'newMockScene',
};

const mockMeetingTable = {
  title: 'mockTitle',
  type: 'RoundMeetingTable',
  logoUrl: 'https://via.placeholder.com/150',
  zoomUrl: 'https://zoom.us/',
  posX: 0,
  posY: 0,
};

export { mockEventRoom, mockEventRoomUpdate, mockMeetingTable };
