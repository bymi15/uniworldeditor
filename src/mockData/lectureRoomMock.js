const mockLectureRoom = {
  title: 'mockTitle',
  module: 'mockModule',
  lecturer: 'mockLecturer',
  startTime: 'mockStartTime',
  endTime: 'mockEndTime',
  firstSlideUrl: 'mockUrl',
  numSlides: 1,
};

const mockLectureRoomUpdate = {
  title: 'newMockTitle',
  module: 'newMockModule',
  lecturer: 'newMockLecturer',
  startTime: 'newMockStartTime',
  endTime: 'newMockEndTime',
  firstSlideUrl: 'newMockUrl',
  numSlides: 3,
};

export { mockLectureRoom, mockLectureRoomUpdate };
