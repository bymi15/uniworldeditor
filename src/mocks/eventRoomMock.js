const mockEventRoom = {
  title: 'mockTitle',
  eventDate: new Date().toISOString(),
  meetingTables: [],
  background: 'mockBackground',
  scene: 'mockScene',
};

const mockEventRoomUpdate = {
  title: 'newMockTitle',
  eventDate: new Date().toISOString(),
  meetingTables: [],
  background: 'newMockBackground',
  scene: 'newMockScene',
};

const mockMeetingTable = {
  title: 'mockTitle',
  type: 'RoundMeetingTable',
  logoUrl: 'https://via.placeholder.com/150',
  zoomUrl: 'https://zoom.us/',
  posX: 0,
  posY: 0,
};

export { mockEventRoom, mockEventRoomUpdate, mockMeetingTable };
