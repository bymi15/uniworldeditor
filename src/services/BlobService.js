import { axiosConfig } from "../config";

class BlobService {
  url = "/blobs";

  async get(container) {
    return (await axiosConfig.get(`${this.url}/${container}`)).data.blobs;
  }

  async upload(data, container) {
    return (await axiosConfig.post(`${this.url}/${container}`, data)).data.url;
  }
}

export default new BlobService();
