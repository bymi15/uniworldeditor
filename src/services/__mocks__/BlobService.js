const BlobService = {
  get: jest.fn(() => Promise.resolve({ test: 'hello' })),
  upload: jest.fn(Promise.resolve()),
  delete: jest.fn(Promise.resolve()),
  deleteSlides: jest.fn(Promise.resolve()),
};

export default BlobService;
