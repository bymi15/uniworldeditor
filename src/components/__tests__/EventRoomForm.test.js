import React from 'react';
import { shallow } from 'enzyme';
import EventRoomForm from '../EventRoomForm';
import { mockEventRoomUpdate } from '../../mockData/eventRoomMock';
import BlobService from '../../services/BlobService';
jest.mock('../../services/BlobService');

describe('EventRoomForm test', () => {
  const mockUseEffect = () => {
    jest.spyOn(React, 'useEffect').mockImplementation((f) => f());
  };

  it('should match snapshot', () => {
    const component = shallow(
      <EventRoomForm
        onSubmit={() => {}}
        submitText='mockText'
        updateEventRoom={mockEventRoomUpdate}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should fetch backgrounds from blob service on load', () => {
    mockUseEffect();
    shallow(<EventRoomForm onSubmit={() => {}} submitText='mockText' />);
    expect(BlobService.get).toBeCalledWith('backgrounds');
  });
});
