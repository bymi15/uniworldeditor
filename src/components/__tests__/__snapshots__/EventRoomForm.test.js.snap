// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventRoomForm test should match snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={2}
>
   
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={12}
  >
    <WithStyles(ForwardRef(Card))
      className="makeStyles-eventInfoPanel-1"
    >
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Grid))
          container={true}
          spacing={3}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            md={5}
            xs={12}
          >
            <WithStyles(ForwardRef(TextField))
              InputLabelProps={
                Object {
                  "shrink": true,
                }
              }
              disabled={false}
              fullWidth={true}
              label="Title"
              name="title"
              onChange={[Function]}
              placeholder="Enter the event title"
              value=""
            />
            <MuiPickersUtilsProvider
              utils={[Function]}
            >
              <PickerWithState
                KeyboardButtonProps={
                  Object {
                    "aria-label": "change date",
                  }
                }
                allowKeyboardControl={true}
                autoOk={true}
                disableToolbar={true}
                disabled={false}
                format="YYYY-MM-DD"
                fullWidth={true}
                id="eventDate"
                invalidDateMessage="Invalid Date Format"
                label="Event Date"
                margin="normal"
                maxDate={2100-01-01T00:00:00.000Z}
                maxDateMessage="Date should not be after maximal date"
                minDate={1900-01-01T00:00:00.000Z}
                minDateMessage="Date should not be before minimal date"
                onChange={[Function]}
                openTo="date"
                value="2021-03-18T01:03:12.275Z"
                variant="inline"
                views={
                  Array [
                    "year",
                    "date",
                  ]
                }
              />
            </MuiPickersUtilsProvider>
            <div
              className="makeStyles-divider-2"
            />
            <WithStyles(ForwardRef(Autocomplete))
              disableClearable={true}
              fullWidth={true}
              getOptionLabel={[Function]}
              onChange={[Function]}
              options={
                Array [
                  "Default",
                  "ConferenceHall",
                  "LaboratoryHall",
                ]
              }
              renderInput={[Function]}
              value="Default"
            />
            <div
              className="makeStyles-divider-2"
            />
            <WithStyles(ForwardRef(Grid))
              className="makeStyles-flexDisplay-9"
              item={true}
              xs={true}
            >
              <WithStyles(ForwardRef(Autocomplete))
                className="makeStyles-padRight-8"
                disableClearable={true}
                fullWidth={true}
                getOptionLabel={[Function]}
                groupBy={[Function]}
                onChange={[Function]}
                options={Array []}
                renderInput={[Function]}
                value="Default"
              />
              <FileUpload
                accept="image/*"
                name="backgroundUpload"
                onChange={[Function]}
                width="120px"
              >
                Upload
              </FileUpload>
            </WithStyles(ForwardRef(Grid))>
            <div
              className="makeStyles-divider-2"
            />
            <WithStyles(ForwardRef(Grid))
              container={true}
              spacing={4}
            >
              <WithStyles(ForwardRef(Grid))
                item={true}
                xs={6}
              >
                <WithStyles(ForwardRef(Typography))
                  variant="subtitle2"
                >
                  Scene Preview
                </WithStyles(ForwardRef(Typography))>
                <div
                  style={
                    Object {
                      "position": "relative",
                    }
                  }
                >
                  <div
                    className="makeStyles-circle-10"
                  />
                  <img
                    alt="background preview"
                    width="100%"
                  />
                </div>
              </WithStyles(ForwardRef(Grid))>
              <WithStyles(ForwardRef(Grid))
                item={true}
                xs={6}
              >
                <WithStyles(ForwardRef(Typography))
                  variant="subtitle2"
                >
                  Â 
                </WithStyles(ForwardRef(Typography))>
                <WithStyles(ForwardRef(Typography))
                  component="h3"
                  variant="subtitle2"
                >
                  Scene:
                </WithStyles(ForwardRef(Typography))>
                <WithStyles(ForwardRef(Typography))
                  component="h3"
                  variant="caption"
                >
                  Default
                </WithStyles(ForwardRef(Typography))>
                <WithStyles(ForwardRef(Typography))
                  className="makeStyles-padTop-7"
                  component="h3"
                  variant="subtitle2"
                >
                  Background:
                </WithStyles(ForwardRef(Typography))>
                <WithStyles(ForwardRef(Typography))
                  component="h3"
                  variant="caption"
                >
                  Default
                </WithStyles(ForwardRef(Typography))>
              </WithStyles(ForwardRef(Grid))>
            </WithStyles(ForwardRef(Grid))>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            md={7}
            xs={12}
          >
            <WithStyles(ForwardRef(Typography))
              variant="subtitle2"
            >
              Meeting Tables
            </WithStyles(ForwardRef(Typography))>
            <div
              className="makeStyles-divider-2"
            />
            <div
              className="makeStyles-centerTableGrid-3"
            >
              <TableGridEditor
                tables={Array []}
                updateMeetingTables={[Function]}
              />
            </div>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(CardActions))>
        <WithStyles(ForwardRef(Button))
          className="makeStyles-createButton-4"
          color="primary"
          disabled={false}
          onClick={[Function]}
          size="large"
          variant="outlined"
        >
          <Memo([object Object])
            className="makeStyles-addIcon-5"
          />
          mockText
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(CardActions))>
    </WithStyles(ForwardRef(Card))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
